src/oscar/core/filestorage.py

```diff
@@ -0,0 +1,46 @@
+from django.contrib.staticfiles.storage import ManifestStaticFilesStorage
+
+module_patterns = (
+    ('*.js', (
+        (
+            r"""
+                (^|;)\s*(?P<matched>(?P<prefix>import(\s*\*\s*as\s+\w+\s+|\s+\w+\s+|\s*{[\w,\s]*?}\s*)from\s*)['"](?P<url>.*?)['"])
+            """,
+            "%(prefix)s '%(url)s'",
+        ),
+    )),
+)
+
+
+def get_static_url(path):
+    from django.contrib.staticfiles.storage import staticfiles_storage
+
+    return staticfiles_storage.url(path)
+
+
+class NoSourceMapsStorage(ManifestStaticFilesStorage):
+    """
+    ManifestStaticFilesStorage with additional features:
+    1. If you have a manifest files error, the page to render the error probably
+    has the same error, and then you can't debug it. This returns the error
+    string so one can see in the rendered html there was a problem
+    2. Remove the sourcemap patterns, add javascript module support.
+    """
+
+    # ------ Temp until: https://code.djangoproject.com/ticket/33353#comment:11
```

Comment:
I don't think this is the right solution for the problem, especially not as a default setting in (`oscar/defaults.py`) because most people will use a different storage backend such has WhiteNoise.

I see two fixes (as the django ticket also points out):
1. Add the .map files to the static folder (bootstrap has them)
2. Remove the `//# sourceMappingURL=bootstrap.bundle.min.js.map` (at the bottom of the file) from the JS file, the collectstatic command will just work fine then.

It doesn't seem like they're gonna change this behaviour.